hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/DependencyMigrationTest.java

```diff
@@ -0,0 +1,226 @@
+            Assertions.assertThatThrownBy(() ->
+                            subject.doMigrations(merkleTree, SemanticVersion.DEFAULT, null, versionedConfig, null))
+                    .isInstanceOf(NullPointerException.class);
+        }
+    }
+
+    @Test
+    @DisplayName("Inter-service dependency migration works")
+    void genesisWithNullVersion() {
+        // Given: we register the EntityIdService, followed by our DependentService
+        final var servicesRegistry =
+                new ServicesRegistryImpl(mock(ConstructableRegistry.class), new NoOpGenesisRecordsBuilder());
+        final DependentService dsService = new DependentService();
+        Set.of(entityServiceStartingWithInitialId(), dsService).forEach(servicesRegistry::register);
+
+        // When: the migrations are run
+        final var subject = new OrderedServiceMigrator(servicesRegistry, mock(ThrottleAccumulator.class));
+        subject.doMigrations(
+                merkleTree,
+                SemanticVersion.newBuilder().major(2).build(),
+                null,
+                mock(VersionedConfigImpl.class),
+                networkInfo);
+
+        // Then: we verify the migrations had the desired effects on both entity ID state and DependentService state
+        // First check that the entity ID service has an updated entity ID, despite its schema migration not doing
+        // anything except setting the initial entity ID. DependentService's schema #2 should have caused the increments
+        // with its new additions to its own state.
+        final var postMigrationEntityIdState =
+                merkleTree.createReadableStates(EntityIdService.NAME).getSingleton(EntityIdService.ENTITY_ID_STATE_KEY);
+        assertThat(postMigrationEntityIdState.get()).isEqualTo(new EntityNumber(INITIAL_ENTITY_ID + 2));
+
+        // Also verify that both of the DependentService's schema migrations took place. First the initial mappings are
+        // created, then the new mappings dependent on the incrementing entity ID are added
+        final var postMigrationDsState =
+                merkleTree.createReadableStates(DependentService.NAME).get(DependentService.STATE_KEY);
+        assertThat(postMigrationDsState.get(INITIAL_ENTITY_ID - 1)).isEqualTo("previously added");
+        assertThat(postMigrationDsState.get(INITIAL_ENTITY_ID)).isEqualTo("last added");
+        assertThat(postMigrationDsState.get(INITIAL_ENTITY_ID + 1)).isEqualTo("newly-added 1");
+        assertThat(postMigrationDsState.get(INITIAL_ENTITY_ID + 2)).isEqualTo("newly-added 2");
+    }
+
+    private EntityIdService entityServiceStartingWithInitialId() {
+        return new EntityIdService() {
+            @Override
+            public void registerSchemas(@NonNull SchemaRegistry registry) {
+                registry.register(new Schema(SemanticVersion.DEFAULT) {
+                    @NonNull
+                    @Override
+                    public Set<StateDefinition> statesToCreate() {
```

Comment:
IntelliJ complains that `StateDefinition` is a raw type (doesn't have generic parameters `K` and `V`) - but then, that's what the declaration in `Schema` does with a "rawtype" suppress.  I guess IntelliJ isn't sophisticated enough to respect _here_ at the override the suppress given _there_ at the decl.

Reply:
🤷 sophistry isn't everything I guess